application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/ConfigurationDataProvider.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.data;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.InitializationException;
+
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+import com.xwiki.admintools.internal.util.DefaultFileOperations;
+
+/**
+ * Extension of {@link AbstractDataProvider} for retrieving configuration data.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named(ConfigurationDataProvider.HINT)
+@Singleton
+public class ConfigurationDataProvider extends AbstractDataProvider
+{
+    /**
+     * The hint for the component.
+     */
+    public static final String HINT = "configuration";
+
+    private final String template = "data/configurationTemplate.vm";
+
+    private Map<String, String> supportedDB;
+
+    @Inject
+    private DefaultFileOperations fileOperations;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        supportedDB = new HashMap<>();
+        supportedDB.put("mysql", "MySQL");
+        supportedDB.put("hsqldb", "HSQLDB");
+        supportedDB.put("mariadb", "MariaDB");
+        supportedDB.put("postgresql", "PostgreSQL");
+        supportedDB.put("oracle", "Oracle");
+    }
+
+    @Override
+    public String getIdentifier()
+    {
+        return HINT;
+    }
+
+    @Override
+    public String provideData()
+    {
+        Map<String, String> systemInfo = generateJson();
+        if (systemInfo != null) {
+            systemInfo.put(serverFound, "found");
+        } else {
+            systemInfo = new HashMap<>();
```

Comment:
Maybe you could return an empty map from `generateJSON`? And check that the map is empty, instead of checking for null

Reply:
Modified the function to throw an exception in case an error is met.